internal/consume/RegistryProtobufMessageDeserializer.go

```diff
@@ -146,17 +148,23 @@ func (deserializer RegistryProtobufMessageDeserializer) schemaToFileDescriptor(s
 	return parseFileDescriptor(".", dependencies)
 }
 
-func (deserializer RegistryProtobufMessageDeserializer) resolveDependencies(references []srclient.Reference) (map[string]string, error) {
-	resolved := map[string]string{}
+func (deserializer RegistryProtobufMessageDeserializer) resolveDependencies(resolved map[string]string, references []srclient.Reference) error {
 	for _, r := range references {
+		if _, ok := resolved[r.Name]; ok {
+			continue
+		}
 		latest, err := deserializer.registry.GetLatestSchema(r.Subject)
```

Comment:
that should better be `deserializer.registry.GetSchemaByVersion(r.Subject, r.Version)` right?

Reply:
Yes. You're absolutely right. I'll change that